formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/conversion/ProgramConverter.kt

```diff
@@ -78,17 +79,26 @@ class ProgramConverter(
     override val anonVarProducer = FreshEntityProducer(::AnonymousVariableEmbedding)
     override val anonBuiltinVarProducer = FreshEntityProducer(::AnonymousBuiltinVariableEmbedding)
     override val returnTargetProducer = FreshEntityProducer(::ReturnTarget)
+    override val nameResolver = SimpleNameResolver()
 
     val program: Program
         get() = Program(
             domains = listOf(RuntimeTypeDomain(classes.values.toList())),
             // We need to deduplicate fields since public fields with the same name are represented differently
             // at `FieldEmbedding` level but map to the same Viper.
             fields = SpecialFields.all.map { it.toViper() } +
-                    fields.distinctBy { it.name.mangled }.map { it.toViper() },
+                    with(nameResolver) { fields.distinctBy { it.name.mangled }.map { it.toViper() } },
             functions = SpecialFunctions.all,
-            methods = SpecialMethods.all + methods.values.mapNotNull { it.viperMethod }.distinctBy { it.name.mangled },
-            predicates = classes.values.flatMap { listOf(it.details.sharedPredicate, it.details.uniquePredicate) }
+            methods = SpecialMethods.all + with(nameResolver) {
```

Comment:
I'd recommend using the debug name mangling for this, so that it isn't resolver-dependent.  It'll make later test changes easier.